src/os/bluestore/BlueStore.h

```diff
@@ -1826,8 +1828,97 @@ class BlueStore : public ObjectStore,
       trace_period_mcs = rate > 0 ? floor((1/rate) * 1000000.0) : 0;
 #endif
     }
+    int64_t get_kv_throttle_current() {
+      return throttle_bytes.get_current();
+    }
+    void reset_kv_throttle_max(int64_t m) {
+      throttle_bytes.reset_max(m);
+    }
   } throttle;
 
+  /**
+   * CoDel algorithm to calculate the incoming batch size to kv by using Throttle mechanism
+   */
+  class BlueStoreSlowFastCoDel {
+  public:
+    BlueStoreSlowFastCoDel(CephContext *_cct);
+
+    ~BlueStoreSlowFastCoDel();
+
+    void reset(CephContext *cct);
+
+    void set_throttle(BlueStoreThrottle *_throttle);
+
+    void submit_txc_info(TransContext *txc);
+
+    uint64_t get_bluestore_budget();
+
+    bool is_activated();
+  protected:
+    bool activated = false;    // true if SlowFastCodel is activated
+    int64_t initial_fast_interval;     // Initial interval for fast loop (original codel)
```

Comment:
This and several other variables do not seem to get initialized either here or in the constructor colon initializer list.  Every one of these members must be initialized.